awx/ui_next/src/components/Lookup/shared/reducer.js

```diff
@@ -0,0 +1,92 @@
+// import { useReducer, useEffect } from 'react';
+
+export default function reducer(state, action) {
+  switch (action.type) {
+    case 'SELECT_ITEM':
+      return selectItem(state, action.item);
+    case 'DESELECT_ITEM':
+      return deselectItem(state, action.item);
+    case 'TOGGLE_MODAL':
+      return toggleModal(state);
+    case 'CLOSE_MODAL':
+      return closeModal(state);
+    case 'SET_MULTIPLE':
+      return { ...state, multiple: action.value };
+    case 'SET_VALUE':
+      return { ...state, value: action.value };
+    case 'SET_SELECTED_ITEMS':
+      return { ...state, selectedItems: action.selectedItems };
+    default:
+      throw new Error(`Unrecognized action type: ${action.type}`);
```

Comment:
Not sure if this message gets displayed to the user.  If so, we may need to explore ways of getting it translated.